comm/communicator.go

```diff
@@ -75,6 +75,50 @@ func NewCommunicator(in io.ReadCloser, db db.DB) *Communicator {
 
 		},
 	})
+	comm.shell.AddCmd(&ishell.Cmd{
+		Name: "prefix",
+		Help: "prefix <bucket> <prefix>: show list of keys with given prefix",
+		Completer: func(args []string) []string {
+			if len(args) == 0 {
+				return comm.db.ListBuckets()
+			}
+			if len(args) == 1 {
+				return comm.db.PrefixScan(args[0], args[1], 0, 100)
+			}
+			return nil
+		},
+		Func: func(c *ishell.Context) {
+			nutsCmd, err := comm.reader.Read(c.Cmd.Name, c.Args)
+			if err != nil {
+				c.Err(err)
+				return
+			}
+			value := comm.db.PrefixScan(nutsCmd.bucket, nutsCmd.prefix, 0, 100)
+			comm.printLines(value)
+		},
+	})
+	comm.shell.AddCmd(&ishell.Cmd{
+		Name: "regex",
+		Help: "regex <bucket> <regex>: show list of keys matching given regex",
+		Completer: func(args []string) []string {
+			if len(args) == 0 {
+				return comm.db.ListBuckets()
+			}
+			if len(args) == 1 {
+				return comm.db.PrefixSearchScan(args[0], args[1], 0, 100)
```

Comment:
I think it makes no sense to autocomplete starting with regex. Maybe just use same prefix scan as for "prefix"? And if so, makes sense to extract this function for reuse

Oh, and "get" already uses PrefixScan for `Completer`. Maybe all three is one completer func?

Reply:
Yeah, this is a bit of a nuisance. I've changed the completer to search using PrefixSearchScan but it doesn't work as I expected. Any ideas why?
Nevertheless every command should do something different so it makes no sense to pull them together.

Reply:
Why: abiosoft/ishell uses for completion only those words that **start with** exactly what is in last arg: `https://github.com/abiosoft/ishell/blob/1b6ad7eb4d5ed63f64de5f937427ed5bf8056e4b/completer.go#L36-L38`

Reply:
But it is fine, works on simple cases, 👍